File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        int n=nums.size();
        int farthest=0 , current=0 , jumps=0;
        
        for( int i=0 ; i<n-1 ; i++ ){
            farthest = max( farthest , nums[i]+i );
            if( i == current ){
                current = farthest;
                jumps++;
            }
        }
        return jumps;
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int n= nums.length;
        int farthest=0, current=0, jump=0;
        for( int i=0 ; i<n-1 ; i++ ){
            farthest = Math.max( farthest , nums[i]+i );
            if( i == current ){
                current = farthest;
                jump++;
            }
        }
        return jump;        
    }
}